subsys/bluetooth/mesh/gen_onoff_srv.c

```diff
@@ -80,12 +80,17 @@ static void onoff_set(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
 
 	if (buf->len == 2) {
 		model_transition_buf_pull(buf, &transition);
+		set.transition = &transition;
+	} else if (bt_mesh_dtt_srv_transition_get(srv->model, &transition)) {
+		set.transition = &transition;
 	} else {
-		bt_mesh_dtt_srv_transition_get(srv->model, &transition);
+		/* When Generic OnOff Server is a base model for Light LC
```

Comment:
Hm. Might still DTT be present as part of another complex model? For example, the Scene Setup server extends the Generic Default Transition Time Server model. The specification does not prohibit keeping Scene and LC on the same element.

I guess it requires a more complex analysis of the binary tree of extensions.

Reply:
Thanks for noticing that! I've proposed another solution. Please check the latest commit.

Reply:
This function doesn't depend on extension bindings at all, it will search through this element's models to find the DTT server. If it's present on the element, this function will find it.